vm/src/function.rs

```diff
@@ -35,6 +35,17 @@ impl From<PyObjectRef> for PyFuncArgs {
     }
 }
 
+impl From<(&Args, &KwArgs)> for PyFuncArgs {
+    fn from(arg: (&Args, &KwArgs)) -> Self {
+        let Args(args) = arg.0;
+        let KwArgs(kwargs) = arg.1;
+        PyFuncArgs {
+            args: args.clone(),
+            kwargs: kwargs.iter().map(|(k, v)| (k.clone(), v.clone())).collect(),
```

Comment:
I think you could use `.cloned` here. Clippy wpuld say so, I assume?

Reply:
I don't think it can work through the tuple:

```
44 |             kwargs: kwargs.iter().cloned().collect(),
   |                                   ^^^^^^ expected tuple, found reference
```